warehouse/csp.py

```diff
@@ -48,8 +48,13 @@ def content_security_policy_tween(request):
         # We don't want to apply our Content Security Policy to the debug
         # toolbar, that's not part of our application and it doesn't work with
         # our restrictive CSP.
+        # We also want to exclude `/api/explorer` from the CSP, as it uses
+        # `unsafe-eval` and `unsafe-inline` for the Swagger UI.
         policy = _serialize(policy).format(request=request)
-        if not request.path.startswith("/_debug_toolbar/") and policy:
+        if (
+            not request.path.startswith("/_debug_toolbar/")
+            and not request.path.startswith("/api/explorer")
```

Comment:
I'm not certain if this exposes us to any specific vulnerability, but I've subscribed to https://github.com/swagger-api/swagger-ui/issues/7540 to watch to see if/when the specifics get resolved.

Reply:
Is the intention that we would have https://pypi.org/api/explorer? Does it have to be hosted on pypi.org domain or could the swagger docs be generated and served from docs.pypi.org?

Reply:
I think I agree with @ewdurbin -- this should probably live on a separate subdomain at least.